lib/src/main/java/uk/gov/di/ipv/core/library/domain/Address.java

```diff
@@ -2,13 +2,16 @@
 
 import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
 import com.fasterxml.jackson.annotation.JsonInclude;
+import lombok.EqualsAndHashCode;
 import uk.gov.di.ipv.core.library.annotations.ExcludeFromGeneratedCoverageReport;
 
 @JsonInclude(JsonInclude.Include.NON_NULL)
 @JsonIgnoreProperties(ignoreUnknown = true)
 @ExcludeFromGeneratedCoverageReport
+@EqualsAndHashCode
```

Comment:
This is pretty nifty

Reply:
Is it the defining of these methods that is actually fixing the bug? So that the entries in the sets can be compared correctly?

Reply:
@Wynndow - We needed to add this, as when we add the name, address, dob to the set it was adding duplicates.

Java was using the default equals and hashcode method which was not comparing the fields within the objects - just that they matched in memory.

>  Its default implementation simply checks the object references of two objects to verify their equality. By default, two objects are equal if and only if they are refer to the same memory location.

By overriding equals and hashcode, we can ensure just the fields are compared. This had to be done for Name, NameParts, Address, BirthDate domain objects.